src/instrumentation/instrumentation.erl

```diff
@@ -0,0 +1,66 @@
+%%%-------------------------------------------------------------------
+%%% @doc
+%%%  Module that takes care of writing instrumentation to .csv file
+%%% @end
+%%%-------------------------------------------------------------------
+-module(instrumentation).
+-author("pawel").
+
+%% API
+-export([
+    create_ets/0,
+    start_link/0,
+    init_writer/0,
+    notify/1
+]).
+
+-define(MAX_BATCH_SIZE, 10000).
+
+-include("records.hrl").
+
+
+%%%---------------------------
+%% Exported functions
+%%%---------------------------
+
+create_ets() ->
+    ets:new(instrumentation, [named_table, public]),
+    ets:insert(instrumentation, {csv_writer, null}).
+
+start_link() ->
+    {ok, spawn_link(instrumentation, init_writer, [])}.
+
+notify(Notification) ->
+    ets:lookup_element(instrumentation, csv_writer, 2) ! Notification.
+
+
+%%%---------------------------
+%% Writer process routines
+%%%---------------------------
+
+read_blocking(File) ->
+    receive
+        Notification -> read_nonblocking(File, [Notification], 1)
+    end.
+
+read_nonblocking(File, Batch, ?MAX_BATCH_SIZE) ->
+    write_batch(File, Batch);
+read_nonblocking(File, Batch, Size) ->
+    receive
+        Notification -> read_nonblocking(File, [Notification | Batch], Size + 1)
+    after 0 -> write_batch(File, Batch)
+    end.
```

Comment:
Dziwna nazwa funkcji. Jeżeli zbierasz tutaj batcha, to niech nazwa funkcji to odzwierciedla. Rozważyłbym też wywalenie wywoływania write_batch z tej funkcji i przeniesienie tego do głównej pętli procesu